lib/posix/clock.c

```diff
@@ -8,6 +8,15 @@
 #include <posix/time.h>
 #include <posix/sys/types.h>
 
+/*
+ * `k_uptime_get` returns a timestamp based on an always increasing
+ * value from the system start.  To support the `CLOCK_REALTIME`
+ * clock, this `rt_clock_base` records the time that the system was
+ * started.  This can either be set via 'clock_settime', or could be
+ * set from a real time clock, if such hardware is present.
+ */
+static struct timespec rt_clock_base = {0, 0};
```

Comment:
Uninitialized statics go into .bss and get zero-filled automatically per the C language standard.  Giving them an initializer that just contains zeroes is a noop, and if the compiler isn't smart enough to detect that they end up in .data and have to be part of the binary image.  Remove.